### PR TITLE
DEV-1853 Create list of runcontexts from json when specified

### PR DESCRIPTION
**Pull request by pauldwolfe**
Adds a simple json deserializer which takes a local file and creates a list of run
contexts. Will be used when the run_dir argument is null and the run_json argument is
specified.

### CODE REVIEW

File: patient-db/src/main/java/com/hartwig/hmftools/patientdb/clinical/SampleLoader.java [hunk 1]
```diff
@@ -53,7 +53,35 @@ public SampleLoader(@NotNull final Lims lims) {
     }
 
     @NotNull
-    private static List<RunContext> loadRunContexts(@NotNull String runsDirectory, @NotNull String pipelineVersionFile) throws IOException {
+    private List<RunContext> extractRunContextsFromDirectoryOrJson(final @NotNull ClinicalAlgoConfig config, final String runsDirectory,
+            final String runsJson) throws IOException {
+        List<RunContext> runContexts;
+        if (runsDirectory != null) {
+            String pipelineVersionFile = config.pipelineVersionFile();
```
Comment (nedleitch) [inline patient-db/src/main/java/com/hartwig/hmftools/patientdb/clinical/SampleLoader.java:60]: Given this is a new addition to an existing command line, should we consider also adding logging if both `runsDirectory` and `runsJson` are given (to tell the user that the JSON will be ignored essentially)?

File: patient-db/src/main/java/com/hartwig/hmftools/patientdb/clinical/context/ProductionRunContextFactory.java [hunk 2]
```diff
@@ -20,4 +20,5 @@ public static RunContext fromRunDirectory(@NotNull String runDirectory, @NotNull
         }
         return runContextFromMetaData;
     }
+
```
Comment (nedleitch) [inline patient-db/src/main/java/com/hartwig/hmftools/patientdb/clinical/context/ProductionRunContextFactory.java:23]: Nit: this probably shouldn't have been added

File: patient-db/src/test/java/com/hartwig/hmftools/patientdb/clinical/readers/RunsJsonReaderTest.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Test(expected = IllegalArgumentException.class)
+    public void throwsIllegalArgumentOnEmptyFile() {
+        RunsJsonReader.extractRunContexts(testJson("empty.json"));
+    }
+
+    @Test
+    public void throwsIllegalArgumentOnUnparseableFile() {
+        RunsJsonReader.extractRunContexts(testJson("missing_fields.json"));
+    }
```
Comment (nedleitch) [inline patient-db/src/test/java/com/hartwig/hmftools/patientdb/clinical/readers/RunsJsonReaderTest.java:32]: This looks to be missing the `expected =` attribute. And I don't see the JSON file?

**Review by nedleitch — CHANGES_REQUESTED** [review]

File: patient-db/src/main/java/com/hartwig/hmftools/patientdb/clinical/SampleLoader.java (hunk 1 above)
Comment (pauldwolfe) [reply patient-db/src/main/java/com/hartwig/hmftools/patientdb/clinical/SampleLoader.java:59]: yes lets throw when neither are specified or both

File: patient-db/src/test/java/com/hartwig/hmftools/patientdb/clinical/readers/RunsJsonReaderTest.java (hunk 3 above)
Comment (pauldwolfe) [reply patient-db/src/test/java/com/hartwig/hmftools/patientdb/clinical/readers/RunsJsonReaderTest.java]: meant to remove this test

**Review by nedleitch — APPROVED** [review]

**Pull request merged**